### ISSUE TITLE
Introduce a new Code of Conduct on Discord

### ISSUE THREAD

**Issue by endigo9740** [skeletonlabs/skeleton#780]
Currently we use the CoC that Svelte provides for their own Discord. I'd like to take that, modify it, and ensure it's valid and available for both GitHub and Discord. We're going to enforce the same standards of practice here.

It might be time to revamp the Welcome channel on Discord to be more useful as well.

**Comment by endigo9740** [comment]
@niktek I'm going to get all core team folks involved on this one, but you're the only one I could tag!

**Comment by endigo9740** [comment]
@Sarenor I welcome your help on this as well!

**Comment by Sarenor** [comment]
Sure thing!
So I know there's discord bots/plugins/settings that make it so a user has to read and accept the server rules before they can interact with the rest of the server, haven't investigated that yet though since the only service I really manage is for my friends and me with very small userbase.

I could ask around though.
A more preset CoC is definitely a good idea, too!

**Comment by Sarenor** [comment]
Some further testing with the Mee6 bot on a private server: Should work well for us, but there's probably also alternative bots available that do the same.
Can have only 1 role-emoji post and would have to decide if we add a "restricted" role that gets removed when the user clicks the button or an "enabled" role that gets added when the user clicks the emoji button. Stylistic choice and probably irrelevant.
We can also have only 2 role-emoji bindings with the free tier of the bot, but that should be enough

**Comment by endigo9740** [comment]
Yeah I've been through that UX flow on other channels. It can also be a bit disorienting the first time you run into something like that, so we'll need to take this into account. The rules we want to enforce is importing, but for a small growing community any friction point might be a detriment. But definitely something we can grow into.

**Comment by Sarenor** [comment]
As Neew in Discord pointed out we apparantly get the Rules Screening functionality when we turn the server in a community server. UX should be a lot better that way!
https://discord.com/channels/1003691521280856084/1003699523522142399/1062129779916144691

Discord Docs:
https://support.discord.com/hc/en-us/articles/1500000466882-Rules-Screening-FAQ#:~:text=Server%20Admins%20can%20customize%20their,pressing%20Set%20Up%20Membership%20Screening.

**Comment by endigo9740** [comment]
See also: https://carl.gg/

**Comment by endigo9740** [comment]
@Sarenor I'm going to fold this into https://github.com/skeletonlabs/skeleton/issues/781

There's enough overlap and really no reason to keep this standalone. But some useful info here, so I'll reference it.
